ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -263,30 +277,151 @@ void PowerSystemNode::injectFault (const string& fault_name,
-
-      /*
-      // TODO: Unspecified how to handle end of fault profile, which is
-      // unlikely.  For now, reuse the last entry.
-      if (index + m_profile_increment >= sequence.size()) {
-        ROS_WARN_STREAM_ONCE
-          (fault_name << ": reached end of fault profile, reusing last entry.");
-        // Probably unneeded, but makes index explicit.
-        index = sequence.size() - 1;
+      // It's a different fault profile from whatever the previous stored directory was, OR the
+      // previous fault ran its course, so reset index.      
+      // Load a new fault profile from the provided directory.
+      try
+      {
+        ifstream file(current_fault_directory);
+        if (current_fault_directory.substr(current_fault_directory.size() - 4) != ".csv")
+        {
+          throw ERR_INVALID_FILE_NAME;
+        }
+        else if (file.fail())
+        {
+          throw ERR_NO_CSV_FOUND;
+        }
+        if (end_fault_warning_displayed && designated_file == saved_file)
+        {
+          ROS_INFO_STREAM("Reloading " << designated_file << "...");
+        }
+        else if (saved_fault_directory != "N/A")
+        {
+          ROS_INFO_STREAM("Loading " << designated_file << " and unloading " << saved_file << "...");
+        }
+        else
+        {
+          ROS_INFO_STREAM("Loading " << designated_file << "...");
+        }
+        if (loadCustomFaultPowerProfile(current_fault_directory, designated_file))
+        {
+          custom_fault_ready = true;
+          end_fault_warning_displayed = false;
+          saved_fault_directory = current_fault_directory;
+          saved_file = designated_file;
+          index = 0;
+
+          ROS_WARN_STREAM_ONCE("Custom power faults may exhibit unexpected results. Caution is advised.");
+          ROS_INFO_STREAM(saved_file.substr(0, saved_file.size() - 4) << " activated!");
+        }
+        else
+        {
+          // Custom fault failed to load correctly, likely due to invalid CSV formatting.
+          custom_fault_ready = false;
+        }        
       }
-      else index += m_profile_increment;
+      catch (int err_val)
+      {
+        switch(err_val)
+        {
+          case ERR_INVALID_FILE_NAME:
+            ROS_WARN_STREAM("Invalid file name '" << designated_file << "' entered; Deactivate fault and try again.");
+            break;
+          case ERR_NO_CSV_FOUND:
+            ROS_WARN_STREAM("Could not find a CSV in the 'profiles' directory with name '" << designated_file << "'. Deactivate fault and try again.");
+            break;
+          default:
+            break;
+        }
+        custom_fault_ready = false;
+      }
+    }
+    else
+    {
+      // It's the same fault profile as previously stored, meaning the same fault was reactivated.
+      // Skip loading and continue from previous index.
+      ROS_INFO_STREAM(saved_file.substr(0, saved_file.size() - 4) << " re-activated!");
+      custom_fault_ready = true;
```

Comment:
From discussion I read in the PR, shouldn't this case trigger a reload of the fault profile file?

Reply:
(resolved in another comment; reactivation now fully reloads the file)